packages/webpack/src/config/client.js

```diff
@@ -195,7 +195,7 @@ export default class WebpackClientConfig extends WebpackBaseConfig {
       ...options,
       name: this.name
     }
-    const clientPath = path || `${router.base}/__webpack_hmr/${this.name}`
+    const clientPath = pathFromOptions || `${router.base}/__webpack_hmr/${this.name}`
 
     const hotMiddlewareClientOptionsStr =
       `${querystring.stringify(hotMiddlewareClientOptions)}&path=${clientPath}`.replace(/\/\//g, '/')
```

Comment:
Can we instead push `path` to `hotMiddlewareClientOptions` string options and move default value to L190?

Reply:
Hi pi0!
I'd be glad to, but as I said in my PS, there is a [commit](https://github.com/nuxt/nuxt.js/pull/4796/commits/aefba48e743792547979cdc074c09cfe4d6d9569) from @wangxingkai which is titled `HMR client path should not be encoded`. I don't know why it shouldn't (actually I've tested in local and everything seems to work well) but I'm afraid to create an unwanted BC or a regression by doing it. This is why I kept this way of doing. 

If it's ok for you, I'll do it right now!

Reply:
Thanks for precisely checking history @mathieutu. I see but actually there is not any specific reason. The only reason may be if special chars are in path and encoding makes special problems... If you want to keep it as is would you please add a comment above for making it TODO to investigate? Otherwise we can just refactor and if anything was broken and reported revert.

Reply:
/cc @clarkdo any reason you think would be?

Reply:
I suppose it's from fixing unit test instesd of changing unit test, so it should be OK to change as your suggestion.

Reply:
And done!